File: source/real-order/src/main/java/edu/jmu/seven/sys/mapper/CustomerMapper.java
package edu.jmu.seven.sys.mapper;

import edu.jmu.seven.sys.entity.Customer;
import com.baomidou.mybatisplus.core.mapper.BaseMapper;
import org.apache.ibatis.annotations.Mapper;

/**
 * <p>
 *  Mapper 接口
 * </p>
 *
 * @author Seven
 * @since 2021-07-16
 */
@Mapper
public interface CustomerMapper extends BaseMapper<Customer> {

}

File: source/real-order/src/main/kotlin/edu/jmu/seven/mapper/CustomerMapper.kt
package edu.jmu.seven.mapper

import com.baomidou.mybatisplus.core.mapper.BaseMapper
import edu.jmu.seven.entity.Account
import edu.jmu.seven.entity.Customer
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param
import org.apache.ibatis.annotations.Select

/**
 * <p></p>
 * @author github/blackswords
 * @date 2021/07/17/14:55
 */

@Mapper
interface CustomerMapper : BaseMapper<Customer> {

    @Select("select * from customer where c_id = #{uname}")
    fun selectByName(@Param(value = "uname") id: String) : Customer
}